wasm/instructions.cpp

```diff
@@ -171,4 +176,101 @@ std::optional<std::vector<Instruction>> parse(std::istream &is) {
     }
 }
 
+struct InstructionStringifyVisitor {
+    std::string &out;
+
+    void operator()(Block const &t) {
+        out += Block::kMnemonic;
+        out += " ";
+        out += to_string(t.type);
+        out += " ";
+
+        for (Instruction const &i : t.instructions) {
+            out += "\n";
+            out += "\t";
+            std::string subinst = to_string(i);
+
+            out += subinst;
+        }
+
+        out += "\n";
+        out += "end";
+    }
+
+    void operator()(Loop const &t) {
+        out += Loop::kMnemonic;
+        out += " ";
+        out += to_string(t.type);
+        out += " ";
+
+        for (Instruction const &i : t.instructions) {
+            out += "\n";
+            out += "\t";
```

Comment:
This will have the wrong indentation if we have nested loops or blocks. Maybe we can pass the current indentation along in the to_string-calls, or add a member variable to the InstructionStringifyVisitor keeping track of it?